util/cache-middleware/cache.js

```diff
@@ -0,0 +1,50 @@
+const flatCache = require('flat-cache');
+const qs = require('querystring');
+const { getConfig, getParamValues, getPath } = require('./api.js');
+
+const cache = flatCache.load('rootstock-self-service-support');
+
+function getKey(req) {
+  // recompose query string
+  // this is necessary to prevent cache-busting DOS attacks by adding on
+  // additional (unused) query params
+  const path = getPath(req);
+  const params = qs.stringify(getParamValues(req));
+  return `${path}?${params}`;
+}
+
+// gets another cached value for the same API endpoint
+// or returns default values stored in api-config file
+function getDefaultValues(req) {
+  const path = getPath(req);
+  const key = Object.keys(cache.all()).find((key) => key.startsWith(path));
+  return cache.getKey(key);
+}
+
+function get(req) {
+  return cache.getKey(getKey(req)) ?? getDefaultValues(req);
+}
+
+async function update(req) {
+  try {
+    const cacheData = cache.getKey(getKey(req));
+    const { cacheTtl, fetch } = getConfig(req);
+    const ttl = new Date(); // cache time to live
+    ttl.setSeconds(ttl.getSeconds() - cacheTtl);
```

Comment:
I don't think this line will work as expected. [`Date.getSeconds()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getSeconds) only returns the number of seconds for its current minute - so it's value will always be 0 to 59.

I think a more accurate use of the Date object would be to use `Date.now() / 1000` to get current number of seconds since epoch, then do the TTL subtraction, then the conditional logic based on that.

Reply:
No, this works perfectly, I am sure. I tested it hundreds of times

Reply:
I understand your concern, I know what you mean, but it seams, Date class can handle time units subtraction correctly. I verified this. E.g. if subtract a larger number of seconds than a minute has, it knows how to calculate the correct date

Reply:
Date class is smart enough to do time units subtraction correctly. [Look at the documentation.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setSeconds#description) 
Furthermore I added a test for seconds subtraction (see `test/date.test.js`) and it turns out, it works as I expected